### DISCUSSION TITLE
右手の一番右上のキー(基板では8番)が押しっぱなしになってしまう

### DISCUSSION THREAD

**Discussion by mazdamaz** [Yowkees/keyball#380] [Q&A]
先日keyball61を購入し、組み立てを行っていたところ、ファームウェア書き込み後remapでキーを確認しようとしたところで、
右手右上のキー(基板では8番の番号があるところ)がおしっぱなしになっていることに気が付きました。
![IMG_8725](https://github.com/Yowkees/keyball/assets/147130471/9ea60fd1-a2af-451a-a983-8a0c75988da6)
![IMG_8726](https://github.com/Yowkees/keyball/assets/147130471/6bfa802b-b733-4673-9613-2b760ef756f9)
考えられる原因・対策があればご教示いただけますと幸いです。

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

特定キーが押しっぱなしになる症状は、基板の配線の良からぬ部分がショートして起こります。
下写真の赤丸のようにキーソケットの半田付けがパターンからはみ出していると、周辺の配線にショートすることがあるので、こういった部分の余計な半田を除去してみてください。

または、ダイオードの極性間違いや半田不良の可能性もあります。
下記青丸が一番右上のキーに該当するものです。ただ、その他のダイオードによっても、本件の不具合が起こる可能性がありますので、全体的にダイオードの極性を見直していただければ幸いです。

あとは、ProMicroを取り外して、ProMicro裏のスルーホールで隣とショートしているものがないかテスター等でご確認いただければ幸いです。

以上、お手数ですがご確認お願いいたします。

![KB61_shortfix](https://github.com/Yowkees/keyball/assets/84209772/8d2cad26-11d9-4afc-b6eb-87fd747f5cb9)

**Reply by mazdamaz** [reply]
ご回答ありがとうございます。
一度付近のキーソケットとダイオードを取り外し余分な半田を取り除いて、キーの状態を確認しても押しっぱなしになっているようでした。
もしかしたら、ProMicroの裏面でショートしているかもしれません。ただ、外すのがかなり難しそう&大変そうなので、とりあえずはremapでキーの割り当てをせずにkeyball60として扱いたいと思います。（幸いにも右上の端っこなので影響は少なめ）
